### PR TITLE
Sumologic exporter and metric pipelines into OT Collector & agent

### PR DESCRIPTION
**Pull request by mat-rumian**
###### Description

Addition of the [Sumologic exporter](https://github.com/SumoLogic/opentelemetry-collector-contrib/tree/main/exporter/sumologicexporter) to the Tracing OT Collector pipeline. This will add possibility to send received metrics on the OTC receiver (OTLP protocol) side to Sumo. 

---

###### Testing performed

- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

**Comment by mat-rumian** [comment]
@pmm-sumo I'm not sure from where should I take a value for **source_host** tag in the **Sumologic** exporter configuration - it can be set [here](https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/e58fc6b6b284c319f1f1bb918e081414c8b22589/deploy/helm/sumologic/conf/traces/traces.otelcol.conf.yaml)

**Comment by mat-rumian** [comment]
@perk-sumo maybe you will be able to help me to get a single environment variable related to the **metrics** endpoint url. 

`{{ include "kubernetes.sources.envs" (dict "Context" $ctx "Type" "metrics") }}` - this returns all environment variables related to the metrics. I would like to add only one env var which is related to "default-metrics" source.

**Comment by sumo-drosiek** [comment]
@mat-rumian Regarding hostname. I think we should use hostname from exporters ([attribute `net.peer.name`](https://github.com/open-telemetry/opentelemetry-log-collection/blob/cc6179775512a539155a1588ecad794ab8152b27/operator/builtin/input/tcp/tcp.go#L262)), otherwise we should use local hostname, but it is not implemented in exporter righrt now

Regarding metrics: 
You can add helper for single source. For now there is no possibility to extract single source ([related code](https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/deploy/helm/sumologic/templates/_helpers.tpl#L944-L956))

File: deploy/helm/sumologic/values.yaml [hunk 1]
```diff
@@ -2487,6 +2498,47 @@ otelcol:
...
+        source_category: "exporters.sumologic.source_category.replace"
+        ## Desired source name. Useful if you want to override the source name configured for the source.
+        source_name: "exporters.sumologic.source_name.replace"
+        ## Desired host name. Useful if you want to override the source host configured for the source.
+        source_host: "%{host.name}"
+        ## List of regexes for attributes which should be send as metadata
+        metadata_attributes:
+          - k8s.*
+          - host.name
+        ## Is the timeout for every attempt to send data to the backend. Maximum connection timeout is 55s.
```
Comment (pmalek-sumo) [inline deploy/helm/sumologic/values.yaml:2523]:
```suggestion
        ## Timeout for every attempt to send data to Sumo Logic backend. Maximum connection timeout is 55s.
```

File: deploy/helm/sumologic/values.yaml [hunk 2]
```diff
@@ -2487,6 +2498,47 @@ otelcol:
...
+        timeout: 5s
+        retry_on_failure:
+          enabled: true
+          ## Time to wait after the first failure before retrying
+          initial_interval: 5s
+          ## Is the upper bound on backoff
```
Comment (pmalek-sumo) [inline deploy/helm/sumologic/values.yaml:2529]:
```suggestion
          ## Upper bound on backoff
```

File: deploy/helm/sumologic/values.yaml [hunk 3]
```diff
@@ -2487,6 +2498,47 @@ otelcol:
...
+          max_interval: 30s
+          ## Is the maximum amount of time spent trying to send a batch
```
Comment (pmalek-sumo) [inline deploy/helm/sumologic/values.yaml:2531]:
```suggestion
          ## Maximum amount of time spent trying to send a batch
```

Ditto elsewhere

**Comment by sumo-drosiek** [comment]
@mat-rumian LGTM

**Comment by pmalek-sumo** [comment]
@sumo-drosiek @mat-rumian Do we want to merge this in? And include in 2.2?

**Comment by mat-rumian** [comment]
@pmalek-sumo yes, it is needed in case of customer call for metrics from OpenTelemetry instrumentation

File: tests/tracing/static/simple.output.yaml [hunk 4]
```diff
@@ -16,6 +16,28 @@ data:
...
+          initial_interval: 5s
+          max_elapsed_time: 120s
+          max_interval: 30s
+        sending_queue:
+          enabled: false
+          num_consumers: 10
+          queue_size: 5000
+        source_category: 
+        source_host: '%{host.name}'
+        source_name:
```
Comment (pmalek-sumo) [inline tests/tracing/static/simple.output.yaml:39]: Should this be empty?

File: tests/tracing/static/simple.output.yaml [hunk 5]
```diff
@@ -16,6 +16,28 @@ data:
...
+        retry_on_failure:
+          enabled: true
+          initial_interval: 5s
+          max_elapsed_time: 120s
+          max_interval: 30s
+        sending_queue:
+          enabled: false
+          num_consumers: 10
+          queue_size: 5000
+        source_category:
```
Comment (pmalek-sumo) [inline tests/tracing/static/simple.output.yaml:37]: Should this be empty?

File: tests/tracing/static/collection-monitoring-false.output.yaml [hunk 6]
```diff
@@ -16,6 +16,28 @@ data:
...
+        retry_on_failure:
+          enabled: true
+          initial_interval: 5s
+          max_elapsed_time: 120s
+          max_interval: 30s
+        sending_queue:
+          enabled: false
+          num_consumers: 10
+          queue_size: 5000
+        source_category:
```
Comment (pmalek-sumo) [inline tests/tracing/static/collection-monitoring-false.output.yaml:37]: Should this be empty?

File: tests/tracing/static/collection-monitoring-false.output.yaml [hunk 7]
```diff
@@ -16,6 +16,28 @@ data:
...
+          initial_interval: 5s
+          max_elapsed_time: 120s
+          max_interval: 30s
+        sending_queue:
+          enabled: false
+          num_consumers: 10
+          queue_size: 5000
+        source_category: 
+        source_host: '%{host.name}'
+        source_name:
```
Comment (pmalek-sumo) [inline tests/tracing/static/collection-monitoring-false.output.yaml:39]: Should this be empty?

File: deploy/helm/sumologic/templates/traces/otelcol/deployment.yaml [hunk 8]
```diff
@@ -66,7 +66,8 @@ spec:
         - name: GOGC
           value: "80"
 {{- $ctx := .Values -}}
-{{ include "kubernetes.sources.envs" (dict "Context" $ctx "Type" "traces") }}
+{{ include "kubernetes.sources.envs" (dict "Context" $ctx "Type" "traces") | nindent 8 }}
+{{- include "kubernetes.sources.env" (dict "Context" $ctx "Type" "metrics" "Name" "default" ) | nindent 8 }}
```
Comment (pmalek-sumo) [inline deploy/helm/sumologic/templates/traces/otelcol/deployment.yaml:70]: `{{` and `{{`, can we stick to one or the other?

**Comment by sumo-drosiek** [comment]
Isn't that kind of duplication of replacement fluentd by otc?

**Comment by perk-sumo** [comment]
Hm, now that I think of it I think this should be enabled in both tracing otelcol and the metadata one.

**Review by pmm-sumo — APPROVED** [review]

**Review by perk-sumo — APPROVED** [review]
LGTM

**Comment by pmalek-sumo** [comment]
@pmm-sumo Can we add a meaningful description to this PR 🙏 ? And after that let's merge it

**Pull request merged**